### PR TITLE
#131 fix swapped note-on note-off events

### PR DESCRIPTION
**Pull request by bzamecnik**
This fixes the issue #131 (and helps to solve #52). Basically in case there are adjacent notes in the MIDI but at the same tick the note-on precedes note-off it's meaning was to close the previous note and open new note. The library however closed all open notes, including this. It created an empty event and sometimes long running notes that were closed only after encountering a matching note-off.

The solution was to keep the note-on event if it's at the same tick and we already closed some events.

I created a minimal test covering both cases - correct and swapped order. The input MIDI is produces via `mido`. We test both reading and writing.

Test are green and on real data piano rolls it seems the bug is fixed.

### CODE REVIEW

**Comment by bzamecnik** [comment]
Possibly it might be good to write to a temporary directory and also to test whether the note duration is not by chance zero.

File: tests/test_pretty_midi.py [hunk 1]
```diff
@@ -272,3 +273,59 @@ def simple():
...
+        pm_song = pretty_midi.PrettyMIDI(midi_path)
+
+        def extract_notes(pm_track):
+            return np.array([(note.pitch, note.end-note.start) for note
+                             in pm_track.notes])
+
+        expected = np.array([[72, 0.05], [72, 0.05], [74, 0.05], [72, 0.05]])
+        assert np.allclose(expected, extract_notes(pm_song.instruments[0]))
+
+        midi_written_path = 'overlapping_notes_%s_track_written.mid' % kind
```
Comment (craffel) [inline tests/test_pretty_midi.py]: Can you write this to a NamedTemporaryFile instead?

File: pretty_midi/pretty_midi.py [hunk 2]
```diff
@@ -306,20 +306,37 @@ def __get_instrument(program, channel, track, create_new):
                     # Store this as the last note-on location
                     note_on_index = (event.channel, event.note)
                     last_note_on[note_on_index].append((
-                        self.__tick_to_time[event.time],
-                        event.velocity))
+                        event.time, event.velocity))
```
Comment (craffel) [inline pretty_midi/pretty_midi.py]: Any specific reason you are changing from time to ticks here?  I don't think floating point issues would bite us here unless they are nondeterministic :)

File: tests/test_pretty_midi.py (hunk 1 above)
Comment (bzamecnik) [inline tests/test_pretty_midi.py]: Yes. Fixed in 83e8ebd. We reuse the NamedTemporaryFile while is open using `file.seek(0)`, otherwise on `close()` it gets deleted.

File: pretty_midi/pretty_midi.py (hunk 2 above)
Comment (bzamecnik) [inline pretty_midi/pretty_midi.py]: Coming from computer graphics, comparing floats by equality scares me. We always had to use epsilon, otherwise nightmares may happen. When we have the original integers at hand it seems to me that it's safer to use them for comparison.
